index.rst

```diff
@@ -1,26 +1,363 @@
+services such as DataLink-followers, which do not need to be fully VOSI-compliant
+in their own right.
+An example of a stripped-down response model for the call-through might be:
+
+- "yes, available" is represented by a 200 status and an optional plain-text
+  message body, which, if present, is forwarded as an additional ``note``
+  element in the response from the central availability service, and
+- "no, not available" is represented by a failure code (503 might be appropriate)
+  and, again, any plain text in the response body is forwarded as a ``note``.
+
+The choice between the query options could be configured on the central
+availability service, or it could be implemented as a run-time recognition of
+a plain-text or null 200 response as acceptable.
+
+Caching
+^^^^^^^
+
+In order to avoid excessive load on the back-end service, the central availability
+service could cache the underlying service's response for an appropriate period,
+e.g., 60 seconds, and limit the call-through to when the cache has expired.
+
+It is TBD whether the central availability service would *continually* (upon cache
+expiration) ping the underlying services so configured (providing internally useful
+status information and enabling internal alerts if the central service thinks a
+service is up but it reports itself as non-functional), or only in response to an
+external request.
+
+Ingress configuration and possible directory service
+----------------------------------------------------
+
+It is proposed that ``https://(RSP-external-address)/api/X/availability``, the normal
+endpoint for a VOSI service ``X``, be redirected via ingress rule to 
+``https://(availability-service-address)/api/central-availability/X``.
+
+This proposal expresses a preference for, but does not insist on, making the
+central availability service directly available externally at a single endpoint,
+exposing the service-specific endpoints below that externally as well --
+in addition to satisfying the required per-service VOSI-availability endpoints
+by redirecting them to the appropriate URL.
+That is, ``(availability-service-address)`` might be the same as ``(RSP-external-address)``.
+
+If so, then the possibility appears of allowing the base URL
+``https://(availability-service-address)/api/central-availability``
+to serve as a REST-standard directory of services, returning a list of the
+valid endpoints one level down.
+This proposal does not take a position on that.
+
+A&A considerations
+------------------
+
+The VOSI standard states "the availability binding[s] must be available to
+anonymous requests".
+
+In the "naive" implementation model where each service provides its own externally
+accessible availability endpoint, this places each service in the position of 
+handling external data from unauthenticated clients directly.
+This increases the attack surface of the RSP and might require more careful
+service-by-service vetting of their response to availability queries.
```

Comment:
While this is true, the return value of the `/availability` endpoint doesn't depend on anything in the query, so the attack surface is limited to fundamental HTTP parsing problems. I think we have as a goal possibly supporting read-only public queries for some services in the future, which would be much more of a risk than the availability endpoint, so I think we're living with this risk anyway.

In other words, I think the merits of the central service are primarily in the roll-up, status UI, and support for non-IVOA services, and the security benefits are fairly minor.

Reply:
That makes sense to me.  I'd willing to take out this paragraph altogether if that's what you advise.

Reply:
Removed